### PR TITLE
Train experiment with kaggle dataset

### PR DESCRIPTION
**Pull request by bofori20**
# Description
This is a structured experiment that trained on kaggle dataset of 40,000 for accuracy. Results points to an improve accuracy over story squad data.

## Type of change
- [ *] New feature (non-breaking change which adds functionality)

## Change Status
- [*] Complete, tested, ready to review and merge

## Has This Been Tested
- [ *] Not necessary

## Checklist
- [ *] There are no merge conflicts

### CODE REVIEW

File: structured_experiments/2022.02.17.test/experiment.MD [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+At iteration 6665/10000/10000, Mean rms=1.708%, delta=6.109%, char train=33.509%, word train=60.683%, skip ratio=0%,  New worst char error = 33.509 wrote checkpoint.
+test_experiment          | 
+test_experiment          | Finished! Error rate = 32.122
+test_experiment          | Loaded 1/1 lines (1-1) of document data/kaggle-ground-truth/TEST_39432.lstmf
+test_experiment          | num_docs > 0:Error:Assert failed:in file imagedata.cpp, line 658
+test_experiment          | make: *** [Makefile:292: data/kaggle/checkpoints/kaggle_checkpoint] Illegal instruction
+test_experiment          | make: *** Deleting file 'data/kaggle/checkpoints/kaggle_checkpoint'
+test_experiment          | Finished
+TODO: Create Template
```
Comment (JoshuaJAnderson) [inline structured_experiments/2022.02.17.test/experiment.MD]: Line from the template. Might wanna remove it.

File: structured_experiments/2022.02.17.test/experiment.MD [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+# Conclusions
+
+The model produces a character train of 33.5% and word train of 60.68%
+
+# Reproduce
+
+- Run `docker-compose -f docker-compose.yml build train && docker-compose -f structured_experiments/2022.02.01.test/docker-compose.yml up --build train_test_experiment` to train a tesseract model using provided sample of kaggle data
```
Comment (JoshuaJAnderson) [inline structured_experiments/2022.02.17.test/experiment.MD]: Need to change the .yml directory to the one you've made.

**Review by JoshuaJAnderson — CHANGES_REQUESTED** [review]

**Review by jraval — CHANGES_REQUESTED** [review]
Hi, if you could change the template like Josh said and if  you can write a longer description for this test, I see the title but the description in a bit more detail would be helpful.  I think we saw the accuracy it was much better than the baseline, before adding the preprocessed data, I'm pretty sure you ran with the preprocessed data.   Double check with TPM and our mentors but if you write up something short, I think it would be helpful.  It was really good that you did this and we can use this as a jumping off point.  Thanks.

File: .gitattributes [hunk 3]
```diff
@@ -0,0 +1,7 @@
+data/kaggle-ground-truth/*.jpg filter=lfs diff=lfs merge=lfs -text
+data/kaggle-ground-truth/*.csv filter=lfs diff=lfs merge=lfs -text
+*.box filter=lfs diff=lfs merge=lfs -text
```
Comment (nlittlepoole) [inline .gitattributes]: we don't want to be tracking `.box` or `.lstmf` files as these are artifacts generated by tesseract, not training data

**Review by nlittlepoole — CHANGES_REQUESTED** [review]
@bofori20 you need to track less of these files. I'd look at this stackoverflow [question for instructions.](https://stackoverflow.com/questions/12481639/remove-files-from-git-commit)

File: structured_experiments/2022.02.17.test/experiment.MD (hunk 2 above)
Comment (bofori20) [reply structured_experiments/2022.02.17.test/experiment.MD]: Thanks for catching that.

File: structured_experiments/2022.02.17.test/experiment.MD (hunk 1 above)
Comment (bofori20) [reply structured_experiments/2022.02.17.test/experiment.MD]: @nlittlepoole, I have made changes to it. Would like to know your thoughts on how the results was presented

**Review by nlittlepoole — DISMISSED** [review]

**Review by JoshuaJAnderson — DISMISSED** [review]

**Review by james-caldwell1981 — CHANGES_REQUESTED** [review]
I'm going to agree with Niger here on the number of files tracked.
Otherwise, I like your results and methodology.

File: .gitattributes [hunk 4]
```diff
@@ -1,2 +1,6 @@
 data/kaggle-ground-truth/*.jpg filter=lfs diff=lfs merge=lfs -text
 data/kaggle-ground-truth/*.csv filter=lfs diff=lfs merge=lfs -text
+*.gt.txt filter=lfs diff=lfs merge=lfs -text
```
Comment (bofori20) [inline .gitattributes]: @james-caldwell1981 I already made the requested changes.

File: structured_experiments/2022.02.17.test/experiment.MD [hunk 5]
```diff
@@ -22,10 +19,8 @@ test_experiment          | make: *** [Makefile:292: data/kaggle/checkpoints/kagg
 test_experiment          | make: *** Deleting file 'data/kaggle/checkpoints/kaggle_checkpoint'
 test_experiment          | Finished
 
-
 # Conclusions
 The Character error rate is 33.509% and the accuracy rate of the model is 66.5%
 
 # Reproduce
-
 - Run `docker-compose -f docker-compose.yml build train && docker-compose -f structured_experiments/2022.02.17.test/docker-compose.yml up --build train_test_experiment` to train a tesseract model using provided sample of kaggle data
```
Comment (bofori20) [inline structured_experiments/2022.02.17.test/experiment.MD]: @jraval requested changes have been made.

**Review by nlittlepoole — APPROVED** [review]
:shipit:

**Review by jinjahninjah — APPROVED** [review]
:shipit:

**Review by jraval — APPROVED** [review]
Changes requested were minor, I witnessed the training of this model and the results.  I believe it's ready for approval.

**Review by james-caldwell1981 — APPROVED** [review]
This is great progress on training the model as well as addressing the large number of files needed to do so.

**Pull request closed without merging**